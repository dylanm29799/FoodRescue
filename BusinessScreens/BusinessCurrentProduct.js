/*
 *
 * ClassName: BusinessCurrentProduct.js
 *
 * Date: 28/04/2021
 *
 *
 * @author: Dylan Murphy, X17506166
 *
 * @reference : https://www.udemy.com/course/react-native-the-practical-guide/learn/lecture/15674818?start=0#overview
 * @reference : https://docs.expo.io/
 * @reference : https://firebase.google.com/docs/web/setup
 * @reference : https://github.com/wix/react-native-navigation
 *
 *
 */
import React from "react";
import { StyleSheet } from "react-native";

import BusinessFoodCountdownProduct from "..//components/BusinessFoodCountdownProduct";
import BusinessNormalProduct from "..//components/BusinessNormalProduct";
import { ScrollView } from "react-native-gesture-handler";

const BusinessCurrentProduct = (props) => {
  //Displaying each component
  return (
    <ScrollView style={styles.screen}>
      <BusinessNormalProduct />

      <BusinessFoodCountdownProduct />
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  screen: {
    flex: 1,
    width: "100%",
    backgroundColor: "#ffecd2",
  },
});

export default BusinessCurrentProduct;
