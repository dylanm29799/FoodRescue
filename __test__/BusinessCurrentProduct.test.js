// __tests__/BusinessCurrentProduct.test.js

import React from "react";
import BusinessCurrentProduct from "../BusinessScreens/BusinessCurrentProduct";
import renderer from "react-test-renderer";
import { firebaseConfig } from "../config";
import * as firebase from "firebase";

firebase.initializeApp(firebaseConfig);
jest.mock("react-navigation", () => ({
  withNavigation: (Component) => (props) => (
    <Component navigation={{ navigate: jest.fn() }} {...props} />
  ),
}));
test("renders correctly", () => {
  const tree = renderer.create(<BusinessCurrentProduct />).toJSON();
  expect(tree).toMatchSnapshot();
});
