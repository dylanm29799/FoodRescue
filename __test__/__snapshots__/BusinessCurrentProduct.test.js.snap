// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  collapsable={false}
  onGestureHandlerEvent={[Function]}
  onGestureHandlerStateChange={[Function]}
  style={
    Object {
      "backgroundColor": "#ffecd2",
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MonM",
            "fontSize": 42.857142857142854,
            "paddingVertical": 21.428571428571427,
            "textAlign": "center",
          }
        }
      >
        Ordinary Products
      </Text>
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "flex": 1,
            "width": "95%",
          }
        }
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </View>
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MonM",
            "fontSize": 42.857142857142854,
            "paddingVertical": 21.428571428571427,
            "textAlign": "center",
          }
        }
      >
        Food Countdown
      </Text>
      <RCTScrollView
        data={Array []}
        disableVirtualization={false}
        getItem={[Function]}
        getItemCount={[Function]}
        horizontal={true}
        initialNumToRender={10}
        keyExtractor={[Function]}
        maxToRenderPerBatch={10}
        onContentSizeChange={[Function]}
        onEndReachedThreshold={2}
        onLayout={[Function]}
        onMomentumScrollEnd={[Function]}
        onScroll={[Function]}
        onScrollBeginDrag={[Function]}
        onScrollEndDrag={[Function]}
        removeClippedSubviews={false}
        renderItem={[Function]}
        scrollEventThrottle={50}
        stickyHeaderIndices={Array []}
        style={
          Object {
            "alignContent": "center",
            "flex": 1,
            "width": "95%",
          }
        }
        updateCellsBatchingPeriod={50}
        viewabilityConfigCallbackPairs={Array []}
        windowSize={21}
      >
        <View />
      </RCTScrollView>
    </RCTSafeAreaView>
  </View>
</RCTScrollView>
`;
